### PR TITLE
Implement cmd PS. Fix ax{*,j}

### PR DESCRIPTION
**Pull request by jroimartin**
- Add r_core_project_save_rdb for selective project dump
- Implement command PS and Ps using r_core_project_save_rdb
- Fix commands ax* and axj
- Fix r_anal_xrefs_save (it didn't save DB to project file)

### CODE REVIEW

File: libr/core/project.c [hunk 1]
```diff
@@ -272,81 +272,118 @@ R_API char *r_core_project_info(RCore *core, const char *prjfile) {
...
+	if (opts&R_CORE_PRJ_IO_MAPS) {
 		r_core_cmd (core, "om*", 0);
 		r_cons_flush ();
+	}
+	if (opts&R_CORE_PRJ_SECTIONS) {
 		r_str_write (fd, "# sections\n");
 		r_io_section_list (core->io, core->offset, 1);
 		r_cons_flush ();
+	}
+	if (opts&R_CORE_PRJ_META) {
```
Comment (radare) [inline libr/core/project.c]: missing spaces around the `&`

File: libr/core/project.c [hunk 2]
```diff
@@ -272,81 +272,118 @@ R_API char *r_core_project_info(RCore *core, const char *prjfile) {
...
+	}
+	if (r_file_is_directory (prj)) {
+		eprintf ("Error: Target is a directory\n");
+		free (prj);
+		return false;
+	}
+
+	r_core_project_init (core);
+
+	snprintf (buf, sizeof (buf), "%s.d"R_SYS_DIR"xrefs", prj);
```
Comment (radare) [inline libr/core/project.c]: missing spaces around the `R_SYS_DIR`

File: libr/core/project.c [hunk 3]
```diff
@@ -272,81 +272,118 @@ R_API char *r_core_project_info(RCore *core, const char *prjfile) {
...
+	r_cons_singleton ()->is_interactive = true;
+
+	close (fd);
+	free (filename);
+
+	return true;
+}
+
+R_API int r_core_project_save(RCore *core, const char *file) {
+	int ret = true;
```
Comment (radare) [inline libr/core/project.c]: this is bool

File: libr/include/r_core.h [hunk 4]
```diff
@@ -403,6 +403,7 @@ R_API int r_core_project_open(RCore *core, const char *file);
 R_API int r_core_project_cat(RCore *core, const char *name);
 R_API int r_core_project_delete(RCore *core, const char *prjfile);
 R_API int r_core_project_list(RCore *core, int mode);
+R_API int r_core_project_save_rdb(RCore *core, const char *file, int opts);
```
Comment (radare) [inline libr/include/r_core.h]: return must be bool

File: libr/include/r_core.h [hunk 5]
```diff
@@ -438,6 +439,19 @@ R_API void r_core_sysenv_help(const RCore* core);
 #define R_CORE_BIN_ACC_VERSIONINFO 0x10000
 #define R_CORE_BIN_ACC_ALL	0x4FFF
 
+#define R_CORE_PRJ_FLAGS	0x0001
+#define R_CORE_PRJ_EVAL		0x0002
+#define R_CORE_PRJ_IO_MAPS	0x0004
+#define R_CORE_PRJ_SECTIONS	0x0008
+#define R_CORE_PRJ_META		0x0010
+#define R_CORE_PRJ_XREFS	0x0020
+#define R_CORE_PRJ_FCNS		0x0040
```
Comment (radare) [inline libr/include/r_core.h]: use spaces

File: libr/anal/xrefs.c [hunk 6]
```diff
@@ -154,7 +155,7 @@ R_API int r_anal_xrefs_init (RAnal *anal) {
 static int xrefs_list_cb_rad(RAnal *anal, const char *k, const char *v) {
 	ut64 dst, src = r_num_get (NULL, v);
 	if (!strncmp (k, "ref.", 4)) {
-		char *p = strchr (k+4, '.');
+		char *p = strrchr (k, '.');
```
Comment (radare) [inline libr/anal/xrefs.c]: use r_str_rchr

File: libr/core/cmd_project.c [hunk 7]
```diff
@@ -39,6 +39,11 @@ static int cmd_project(void *data, const char *input) {
 			r_cons_printf ("%s\n", file);
 		}
 		break;
+	case 'S':
+		if (input[1]==' ') {
```
Comment (radare) [inline libr/core/cmd_project.c]: spaces around `==`

File: libr/include/r_core.h (hunk 5 above)
Comment (crowell) [inline libr/include/r_core.h]: uh, its tabs in like all the other places in this areas

**Comment by radare** [comment]
Squashed and merged!

**Pull request closed without merging**